schemas/gen.py

```diff
@@ -271,9 +279,10 @@ def get_defaulted_type(schema: Schema, tag: str, type: str) -> Tuple[str, str, b
             pass
         elif type.startswith("**"):
             raise AssertionError(f"{tag} type ({type}) must not be a double pointer")
-        elif default is not None:
+        elif not required:
             raise AssertionError(
-                f"ERROR: {tag} type ({type}) must be a pointer since it can be defaulted!\n"
+                f"ERROR: {schema.golang_title}.{tag} type ({type}) must be a pointer since is "
+                "not required!\n"
```

Comment:
This block is confusing to me (maybe it's just the error message).

Reply:
I'll see if I can't make the logic and the message clearer.

Reply:
How is this for a simulated message (If I remove `HyperparametersV0` from the whitelist above):

```
ERROR: ExperimentConfigV0.hyperparameters type (HyperparametersV0) must be nil-able, 
since it is not required.  This is because `nil` is how we represent values which were not provided 
by the user.  Nilable types are pointers, slices, or maps.

Note: if HyperparametersV0 is a type alias for a map or slice type, like:

    type BindMountsConfigV0 []BindMounts

then you can safely just add "HyperparametersV0" to KNOWN_MAP_OR_SLICE_ALIAS_TYPES
in schemas/gen.py to avoid this error.
```